core/state/pruner/pruner.go

```diff
@@ -186,16 +186,15 @@ func prune(maindb ethdb.Database, stateBloom *stateBloom, middleStateRoots map[c
 	// Note for small pruning, the compaction is skipped.
 	if count >= rangeCompactionThreshold {
 		cstart := time.Now()
-
-		for b := byte(0); b < byte(16); b++ {
+		for b := 0x00; b <= 0xf0; b += 0x10 {
 			var (
-				start = []byte{b << 4}
-				end   = []byte{(b+1)<<4 - 1}
+				start = []byte{byte(b)}
+				end   = []byte{byte(b + 0x10)}
 			)
-			log.Info("Compacting database", "range", fmt.Sprintf("%#x-%#x", start, end), "elapsed", common.PrettyDuration(time.Since(cstart)))
-			if b == 15 {
+			if b == 0xf0 {
 				end = nil
 			}
+			log.Info("Compacting database", "range", fmt.Sprintf("%#x-%#x", start, end), "elapsed", common.PrettyDuration(time.Since(cstart)))
```

Comment:
If you move it below the `if`, you'll end up printing `nil` into hex. Not sure how that behaves.

Reply:
The very last range's end.

Reply:
Right, neither did I, that's why I made the play thingy above
```
0x00-0x10
0x10-0x20
0x20-0x30
0x30-0x40
0x40-0x50
0x50-0x60
0x60-0x70
0x70-0x80
0x80-0x90
0x90-0xa0
0xa0-0xb0
0xb0-0xc0
0xc0-0xd0
0xd0-0xe0
0xe0-0xf0
0xf0-
```

Reply:
Funky